Index: server/bin/openerp-server.py
diff --git a/server/bin/openerp-server.py b/server/bin/openerp-server.py
index 6d5d0a99854977185ae0deb2848d71239e5d5cd4..e5076e95185e31c3a8df843533719ecba1d15b63 100755
--- a/server/bin/openerp-server.py
+++ b/server/bin/openerp-server.py
@@ -186,6 +186,8 @@ if tools.config['xmlrpc']:
     secure = tools.config["secure"]
 
     httpd = netsvc.HttpDaemon(interface, port, secure)
+    port = httpd.server.socket.getsockname()[1]
+    tools.config['port'] = port
 
     xml_gw = netsvc.xmlrpc.RpcGateway('web-services')
     httpd.attach("/xmlrpc", xml_gw)
@@ -197,6 +199,8 @@ if tools.config['netrpc']:
     netport = int(tools.config['netport'])
     netinterface = tools.config["netinterface"]
     tinySocket = netsvc.TinySocketServerThread(netinterface, netport, False)
+    netport = tinySocket.socket.getsockname()[1]
+    tools.config['netport'] = netport
     logger.notifyChannel("web-services", netsvc.LOG_INFO, 
                          "starting NET-RPC service, port %d" % (netport,))
 
Index: server/bin/tools/config.py
diff --git a/server/bin/tools/config.py b/server/bin/tools/config.py
index 34bf3c4a79b26ed91e5456db68c9bd75cae17c66..6751cf2bb852fb5479d39091569ddac779aa53ce 100644
--- a/server/bin/tools/config.py
+++ b/server/bin/tools/config.py
@@ -238,7 +238,7 @@ class configmanager(object):
             keys.extend(['secure_cert_file', 'secure_pkey_file'])
 
         for arg in keys:
-            if getattr(opt, arg):
+            if getattr(opt, arg) or isinstance(getattr(opt, arg), int):
                 self.options[arg] = getattr(opt, arg)
 
         keys = ['language', 'translate_out', 'translate_in', 'debug_mode',
